demo-extended/src/main/java/com/novoda/downloadmanager/demo/extended/batches/BatchesActivity.java

```diff
@@ -42,6 +44,7 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_batches);
+        downloadsUriProvider = DownloadsUriProvider.getInstance();
```

Comment:
:x: why does the demo application have to do this? This is a creating poor public api. Everything should be done through the `DownloadManagerBuilder` and then `DownloadManager ` only.